src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -113,10 +131,41 @@ public IEnumerable<string> GetManifestDirectories()
                 {
                     if (!IsManifestIdOutdated(workloadManifestDirectory))
                     {
-                        yield return workloadManifestDirectory;
+                        manifestIdsToDirectories.Add(Path.GetFileName(workloadManifestDirectory), workloadManifestDirectory);
                     }
                 }
             }
+
+            if (_knownManifestIds != null && _knownManifestIds.Any(id => !manifestIdsToDirectories.ContainsKey(id)))
+            {
+                var missingManifestIds = _knownManifestIds.Where(id => !manifestIdsToDirectories.ContainsKey(id));
+                foreach (var missingManifestId in missingManifestIds)
+                {
+                    var manifestDir = FallbackForMissingManifest(missingManifestId);
+                    manifestIdsToDirectories.Add(missingManifestId, manifestDir);
+                }
+            }
+
+            return manifestIdsToDirectories.Values;
+        }
+
+        private string FallbackForMissingManifest(string manifestId)
+        {
+            var candidateFeatureBands = Directory.GetDirectories(Path.Combine(_sdkRootPath, "sdk-manifests"))
+                .Select(dir => Path.GetFileName(dir))
+                .Where(featureBand => Version.TryParse(featureBand, out _))
+                .Select(featureBand => Version.Parse(featureBand))
+                .Where(featureBand => featureBand < Version.Parse(_sdkVersionBand));
+            var matchingManifestFatureBands = candidateFeatureBands
+                .Where(featureBand => Directory.Exists(Path.Combine(_sdkRootPath, "sdk-manifests", featureBand.ToString(), manifestId)));
+            if (matchingManifestFatureBands.Any())
+            {
+                return Path.Combine(_sdkRootPath, "sdk-manifests", matchingManifestFatureBands.Max()!.ToString(), manifestId);
+            }
+            else
+            {
+                throw new Exception(string.Format(Strings.KnownManifestIdDoesNotExist, manifestId));
```

Comment:
I'm not sure whether we should be throwing an exception here.  Right now a workaround for when your manifests are messed up is to just delete the sdk-manifests folder.  At a minimum, I think we should have a way to suppress the error and ignore the missing manifests.